File: native/ios/Sources/CovenantPathKit/Views/MaintenanceScreen.swift
#if canImport(UIKit)
import SwiftUI

/// Owner-only MAINTENANCE mode lock screen (migration 0056) — shown to everyone EXCEPT the owner while
/// the global switch is ON. Purely UX: the database's RESTRICTIVE RLS already returns no member rows to
/// non-owners during maintenance, so this just replaces the (empty) dashboard with a friendly message.
/// Mirrors the web `MaintenanceGate` "We'll be right back" screen.
struct MaintenanceScreen: View {
    let message: String?

    private var body_message: String {
        let trimmed = message?.trimmingCharacters(in: .whitespacesAndNewlines) ?? ""
        return trimmed.isEmpty
            ? "Covenant Path is briefly down for maintenance. Please check back in a little while."
            : trimmed
    }

    var body: some View {
        VStack(spacing: 14) {
            Image(systemName: "wrench.and.screwdriver.fill")
                .font(.system(size: 44))
                .foregroundStyle(.secondary)
            Text("We'll be right back")
                .font(.title2.bold())
            Text(body_message)
                .font(.callout)
                .foregroundStyle(.secondary)
                .multilineTextAlignment(.center)
                .frame(maxWidth: 420)
        }
        .padding(32)
        .frame(maxWidth: .infinity, maxHeight: .infinity)
        .background(Color(.systemBackground))
        .ignoresSafeArea()
    }
}
#endif

File: native/android/app/src/main/java/org/membercovenantpath/viewer/ui/screens/MaintenanceScreen.kt
package org.membercovenantpath.viewer.ui.screens

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.widthIn
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Build
import androidx.compose.material3.Icon
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Surface
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp

/**
 * Owner-only MAINTENANCE mode lock screen (migration 0056) — shown to everyone EXCEPT the owner while
 * the global switch is ON. Purely UX: the database's RESTRICTIVE RLS already returns no member rows to
 * non-owners during maintenance, so this just replaces the (empty) dashboard with a friendly message.
 * Mirrors the web `MaintenanceGate` + iOS `MaintenanceScreen`.
 */
@Composable
fun MaintenanceScreen(message: String?) {
    Surface(modifier = Modifier.fillMaxSize(), color = MaterialTheme.colorScheme.background) {
        Column(
            modifier = Modifier.fillMaxSize().padding(32.dp),
            horizontalAlignment = Alignment.CenterHorizontally,
            verticalArrangement = Arrangement.Center,
        ) {
            Icon(
                Icons.Filled.Build,
                contentDescription = null,
                modifier = Modifier.size(56.dp),
                tint = MaterialTheme.colorScheme.onSurfaceVariant,
            )
            Spacer(Modifier.height(14.dp))
            Text("We'll be right back", style = MaterialTheme.typography.headlineSmall)
            Spacer(Modifier.height(8.dp))
            Text(
                message?.trim()?.takeIf { it.isNotEmpty() }
                    ?: "Covenant Path is briefly down for maintenance. Please check back in a little while.",
                style = MaterialTheme.typography.bodyMedium,
                color = MaterialTheme.colorScheme.onSurfaceVariant,
                textAlign = TextAlign.Center,
                modifier = Modifier.widthIn(max = 420.dp),
            )
        }
    }
}
